src/lib/logical_query_plan/lqp_utils.hpp

```diff
@@ -208,6 +208,32 @@ void visit_lqp_upwards(const std::shared_ptr<AbstractLQPNode>& lqp, Visitor visi
  */
 std::vector<std::shared_ptr<AbstractLQPNode>> lqp_find_subplan_roots(const std::shared_ptr<AbstractLQPNode>& lqp);
 
+/**
+ * Traverses @param lqp from top to bottom to find all leaf nodes.
+ * @return By default, all leaf nodes are returned.
+ *         Optionally, the template parameter can be set to return leaf nodes of type @tparam LQPNode only.
+ */
+template <class LQPNode = AbstractLQPNode>
+std::vector<std::shared_ptr<LQPNode>> lqp_find_leafs(const std::shared_ptr<AbstractLQPNode>& lqp) {
```

Comment:
Does it have to be leaf nodes? If we open this up to return any node of the defined type, it might open up the function to a more general use. When searching leaf nodes, it doesn't change anything, because StoredTableNodes are leaves by definition.